language/stdlib/modules/libra_account.mvir

```diff
@@ -103,23 +116,65 @@ module LibraAccount {
     // Helper to withdraw `amount` from the given `account` and return the resulting LibraCoin.T
     withdraw_from_account(account: &mut R#Self.T, amount: u64): R#LibraCoin.T {
         let to_withdraw: R#LibraCoin.T;
+
         to_withdraw = LibraCoin.withdraw(&mut move(account).balance, copy(amount));
         return move(to_withdraw);
     }
 
     // Withdraw `amount` LibraCoin.T from the transaction sender's account
     public withdraw_from_sender(amount: u64): R#LibraCoin.T {
+        let sender_account: &mut R#Self.T;
+
+        sender_account = borrow_global<T>(get_txn_sender());
+        if (*&copy(sender_account).delegated_withdrawal_capability) {
+            // The sender has delegated the privilege to withdraw from her account elsewhere--abort.
```

Comment:
Is there any specific reason why we want to disallow the withdraw capability after delegation? Are you worried about sender moving all the funds out after delegating the withdraw capability? If so will burn all the money under sender as gas achieve the same result?

Reply:
Good question. The idea is that for use-cases like cold wallets, you want to *guarantee* that funds can only be withdrawn under certain conditions (e.g., no more than 100 Libra/day). If withdrawal capabilities aren't unique, it's much harder to do this because the account owner can still withdraw via the normal mechanism.

If the account owner wants to retain the ability to withdraw after delegating, they can set up a module that grants access both to them and to others.

The gas thing is a good point. If the delegatee is concerned about that, they can require that the delegator rotates their authentication key to (e.g.) all zeros.